### PR TITLE
Enable authenticity_token

### PR DESCRIPTION
**Pull request by arusa**
Not yet ready. authenticity_token is enabled, but now everything is forbidden (even the start page)

Docs:

- https://sinatrarb.com/protection/authenticity_token
- https://github.com/sinatra/sinatra/blob/main/rack-protection/lib/rack/protection/authenticity_token.rb

### CODE REVIEW

**Comment by arusa** [comment]
We need to add options to configure authenticity_token and update the form.
Maybe we also need CSRF for json?
https://sinatrarb.com/protection/json_csrf
https://github.com/sinatra/sinatra/blob/main/rack-protection/lib/rack/protection/json_csrf.rb

File: config.ru [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
 if ActiveRecord::Base.connection.migration_context.needs_migration?
   raise 'Migrations are pending. Run `rake db:migrate` to resolve the issue.'
 end
 
 use Rack::MethodOverride
-
-require 'rack/protection'
-use Rack::Protection, use: %i[content_security_policy], script_src: "'self'", img_src: "'self'"
+use Rack::Session::Cookie,
+    domain: 'localhost:3000',
```
Comment (arusa) [inline config.ru:13]: hardcoded localhost:3000? Even sinatra runs on :9191 for me ;-)

File: config.ru [hunk 2]
```diff
@@ -1,14 +1,24 @@
...
+use Rack::Session::Cookie,
+    domain: ->(env) { Rack::Request.new(env).host },
+    path: '/',
+    expire_after: 3600 * 24,
+    secret: ENV.fetch('SESSION_SECRET') { SecureRandom.hex(64) }
+use Rack::Protection,
+    use: %i[content_security_policy authenticity_token],
+    permitted_origins: ENV.fetch('URL', nil),
+    script_src: "'self'",
+    script_src_attr: "'self'",
```
Comment (arusa) [inline config.ru:21]: What do we need that for? Mozilla says that's for event handlers like "onclick" (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src-attr)

Do we still have such handlers?

File: config.ru [hunk 3]
```diff
@@ -1,14 +1,24 @@
...
+    script_src_elem: "'self'"
```
Comment (arusa) [inline config.ru:22]: Why do we need that? According to Mozilla script-src-elem and script-src-attr are already covered by script-src (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src)

File: public/javascripts/application.js [hunk 4]
```diff
@@ -22,7 +22,10 @@ function showTooltip() {
...
 }
 
-function genQR(){
-   new QRCode(document.getElementById("qrcode"), document.getElementById("secret-url").value);
+function getAuthenticityToken() {
+  return document.querySelector("meta[name='authenticity_token']")?.content;
 }
 
+document.getElementById("copy-button")?.addEventListener("click", copyToClip);
+document.getElementById("copy-button")?.addEventListener("ouseout", showTooltip);
```
Comment (arusa) [inline public/javascripts/application.js:30]: I think "ouseout" should be mouseout?

File: config.ru (hunk 2 above)
Comment (whotwagner) [reply config.ru]: Nice catch! I removed the script_src-configs and it still seems to work. please test it in your environment too

File: public/javascripts/application.js (hunk 4 above)
Comment (whotwagner) [reply public/javascripts/application.js]: fixed

File: config.ru (hunk 2 above)
Comment (arusa) [reply config.ru]: Everything works for me

**Review by whotwagner — APPROVED** [review]

**Pull request merged**